dapp/src/components/wrap/WrapHomepage.js

```diff
@@ -0,0 +1,320 @@
+      const conversionRate = await displayCurrency(
+        await wousd.convertToAssets(calculateSwapAmounts(1, 18).swapAmount),
+        wousd
+      )
+      setRate(conversionRate)
+    }
+    wrapEstimate()
+    approvalNeeded()
+    calculateRate()
+  }, [inputAmount, wousd, allowances])
+
+  useEffect(() => {
+    // currencies flipped
+    if (previousSwapMode !== swapMode) {
+      localStorage.setItem(lastSelectedSwapModeKey, swapMode)
+      const otherCoinAmount = Math.floor(wrapEstimate * 1000000) / 1000000
+      setInputAmount(Math.floor(otherCoinAmount * 100) / 100)
+    }
+  }, [swapMode])
+
+  const errorMap = [
+    {
+      errorCheck: (err) => {
+        return err.name === 'EthAppPleaseEnableContractData'
+      },
+      friendlyMessage: fbt(
+        'Contract data not enabled. Go to Ethereum app Settings and set "Contract Data" to "Allowed"',
+        'Enable contract data'
+      ),
+    },
+    {
+      errorCheck: (err) => {
+        return err.message.includes(
+          'Failed to sign with Ledger device: U2F DEVICE_INELIGIBL'
+        )
+      },
+      friendlyMessage: fbt(
+        'Can not detect ledger device. Please make sure your Ledger is unlocked and Ethereum App is opened.',
+        'See ledger connected'
+      ),
+    },
+  ]
+
+  const onMintingError = (error) => {
+    if (errorMap.filter((eMap) => eMap.errorCheck(error)).length > 0) {
+      setBuyErrorToDisplay(error)
+    }
+  }
+
+  const swapMetadata = () => {
+    const coinGiven = swapMode === 'mint' ? 'wousd' : 'ousd'
+    const coinReceived = swapMode === 'mint' ? 'ousd' : 'wousd'
+    const swapAmount = inputAmount
+    return {
+      coinGiven,
+      coinReceived,
+      swapAmount,
+    }
+  }
+
+  const onWrapOusd = async () => {
+    analytics.track(
+      swapMode === 'mint' ? 'On Swap to wOUSD' : 'On Swap from wOUSD',
+      {
+        category: 'wrap',
+        label: swapMetadata.stablecoinUsed,
+        value: swapMetadata.swapAmount,
+      }
+    )
+
+    const metadata = swapMetadata()
+
+    try {
+      analytics.track('Before Swap Transaction', {
+        category: 'swap',
+        label: metadata.stablecoinUsed,
+        value: metadata.swapAmount,
+      })
+
+      let result
+      if (swapMode === 'mint') {
+        result = await signer(wousd).deposit(
+          calculateSwapAmounts(inputAmount, 18).swapAmount,
+          account
+        )
+      } else {
+        result = await signer(wousd).redeem(
+          calculateSwapAmounts(inputAmount, 18).swapAmount,
+          account,
+          account
+        )
+      }
+
+      storeTransaction(
+        result,
+        swapMode === 'mint' ? 'redeem' : 'mint',
+        'wousd',
+        {
+          ousd: inputAmount,
+          wousd: inputAmount,
+        }
+      )
+      setStoredCoinValuesToZero()
+      setInputAmount('')
+
+      const receipt = await rpcProvider.waitForTransaction(result.hash)
+      analytics.track('Swap succeeded User source', {
```

Comment:
`Swap succeeded User source` to `Wrap succeeded User source`
and `category: swap` to `category: wrap`